src/minio/managers/user_manager.py

```diff
@@ -490,3 +516,53 @@ async def _delete_user_home_directory(self, username: str) -> None:
                 await self.client.delete_object(bucket_name, obj_key)
 
             logger.info(f"Deleted {len(objects)} objects from {user_prefix}")
+
+    async def _delete_user_system_directory(self, username: str) -> None:
+        """Delete user system directories and all contents (like Spark job logs)."""
+
+        # Get only user-scoped system paths to avoid deleting global resources
+        system_paths = self._get_user_system_paths(username, user_scoped_only=True)
+
+        # Delete directories for each system bucket and prefix
+        for bucket_name, prefixes in system_paths.items():
+            if not await self.client.bucket_exists(bucket_name):
+                continue  # Skip if bucket doesn't exist
+
+            for prefix in prefixes:
+                # List all objects in user system directory
+                objects = await self.client.list_objects(bucket_name, prefix)
```

Comment:
This will only list 10k objects IIUC. If someone uploaded seqs for AllTheBacteria most of them are going to stick around

Reply:
okay - that would be a pretty large refactor - are you sure we will have a case where this becomes an issue?

Reply:
okay - created ticket https://kbase-jira.atlassian.net/browse/CDM-296

Reply:
:+1: